
\section{Introduction}
Biobanks store and catalog human biological material from identifiable individuals for both clinical and research purposes. Recent initiatives in Personalized Medicine created a steeply increasing demand to sequence the human biological material stored in Biobanks. Such large-scale sequencing is no on the way in probably hundreds of projects around the world, reaching a dimension of up-to 100.000 genomes in a single project\footnote{See \url{http://www.genomicsengland.co.uk/}.}. Furthermore, sequencing also is becoming more and more routine in a clinical setting for improving diagnosis and therapy especially in cancer\cite{pmedicine}. However, software systems for Biobanks traditionally managed only metadata associated with samples, such as pseudo-identifiers for patients, sample collection information, or study information. Such systems cannot cope with the current requirement to, alongside such metadata, also store and analyse genomic data, which might mean everything from a few megabytes (e.g. genotype information from a SNP array) to hundreds of Gigabyte per sample (for whole genome sequencing with adequate accuracy). 

For long, such high-throughput sequencing and analysis was only accessible for large research centres that (a) could afford enough modern sequencing devices and (b) were equipped with considerable stuffing to run high performance computing clusters. This situation must now change. Sequencing devices become cheaper and cheaper, and more and more labs and hospitals depend on sequencing information for daily research and diagnosis/treatment. To make this possible, there is a pressing need for flexible and open software systems also enabling the computation analysis of large biomedical data sets at a reasonable price. Note that this trend is not restricted to genome sequencing; very similar developments also happen in other medical areas, such as molecular imaging\cite{imaging}, drug discovery\cite{drug}, or data generated from patient-attached sensors\cite{qself}. 

Here, we present our platform currently under development in a joint team of computer scientists, bioinformaticians, and pathologists. The system is designed as ``platform-as-a-service'', i.e., it is meant to be easily installed on a local cluster (or, equally well, in a public cloud) using Karamel/Chef~\footnote{http://www.karamel.io}. Primary design goals are flexibility in terms of the analysis being performed, scalability up to very large data sets and very large cluster set-ups, ease of use and low maintenance cost, strong support for data security and data privacy, and direct usability for users. To this end, in encompasses (a) a scientific workflow engine running on top of the popular Hadoop platform for distributed computing, (b) a scientific workflow language focusing on easy integration of existing tools and simple rebuilding of existing pipelines, (c) a completely scripted cluster set-up and, and (d) role-based access control. It also features (e) an enhanced distributed file system enhancing throughput, metadata management, and consistency while reducing storage requirements compared to HDFS, (f) CHIRON, which enables the federation of clouds on the file system level, and (g) a simple Laboratory Information Management Service with integrated Web Interface for authenticating/authorizing users, managing data, designing and searching for metadata, and support for running workflows and analysis jobs on Hadoop. This web interface hides much of the complexity of the Hadoop backend, and supports multi-tenancy through first-class support for \textit{Studies}, \textit{SampleCollections} (DataSets), \textit{Samples}, and \textit{Users}. 

In this paper, we give an overview on the overall architecture of the BioBankCloud platform and describe each component in more detail. The entire system is currently under development; while a number of components already have been released for immediate usage, a first overall platform release is planned for the near future. The system is essentially agnostic to the type of data being managed and the types of analysis being performed, but developed with genome sequencing as most important application area. Therefore, throughout this paper we will use examples from this domain. 