\section{Introduction}
Biobanks store and catalog human biological material from identifiable individuals for both clinical and research purposes. Recent advances in Next-Generation Sequencing (NGS) technology has meant that there is an increasing demand to sequence the human biological material stored in Biobanks. Both research projects and clinical health-care systems use Biobanks to store samples that are sequenced using techniques from genotyping to whole-genome sequencing (WGS). Biobanks' computer systems have traditionally managed only metadata associated with samples, such as pseudo-identifiers for patients, sample collection information, study information, and other data concerning samples. Alongside this metadata, we now increasingly need to store genomic data, which requires anything from MBs (genotyping) up to several hundred GBs (WGS) of data per sample.
The data storage requirements for large-scale WGS sequencing projects, such as the 100,000 genomes project by Genomics England, now scale to many PBs, and are so large that researchers are looking at cost-effective Big Data solutions based on commodity hardware, such as Hadoop. When data volumes grow to many TBs, traditional database technology (including sharded relational databases) is no longer viable, as computations on the data neccessitate transfer more data from storage to compute nodes than is practical with current networking technology. The solution provided by platforms such as Hadoop is to move computation to where the data is located, exploiting the principle of \textit{data locality}. That is, jobs are parallelized across many nodes and each job loads its (large) input data primarily from disk subsystems, while network I/O is used in subsequent processing steps with, typically, smaller data volumes (relative to the input data size).

In this paper, we introduce BiobankCloud, an integrated platform, based on Hadoop, for the secure storage, processing, and sharing of genomic data and associated metadata. BiobankCloud is a Platform-as-a-Service that can be automatically deployed on public clouds, private clouds or bare-metal servers using Karamel/Chef~\footnote{http://www.karamel.io}. As part of BiobankCloud, we also provide a Laboratory Information Management Service (LIMS) as software-as-a-service (SaaS). The LIMS has an integrated User Interface (UI) for authenticating/authorizing users, managing data, designing and searching for metadata, and support for running workflows and analysis jobs on Hadoop. The LIMS hides much of the complexity of the Hadoop backend, and supports multi-tenancy through first-class support for \textit{Studies}, \textit{SampleCollections} (DataSets), \textit{Samples}, and \textit{Users}.

In BiobankCloud, we have developed our own Hadoop distribution, Hadoop Open Platform-as-a-Service (Hops), with improved scalability and customizability properties, enabled by a new metadata storage system based on a distributed in-memory database. In addition to Hops, existing scientific workflow management systems are typically custom-built and have not been designed for data parallel processing with data locality. For performance reasons, their architectures and file formats are typically flat (monolithic). None of the main file formats in genomics (fastq, BAM/SAM, CRAM, and VCF) have support for data parallel processing because of their assumption of centralized metadata (e.g., in the file header). In BiobankCloud, we provide SAASFEE, a scientific workflow management system, which includes a native second-level scheduler for YARN and a workflow language which bridges the gap between data-parallel processing and supporting existing file formats and foreign code. SAASFEE can both speed up workflows for a NGS variant calling on 24 machines and scale out to support larger clusters.